### PR TITLE
Issue 631

### LINKED ISSUE #631: Create Default Logging level configuration for ESAPI library
Derived from issue #630, the default logging level of all ESAPI loggers is `Level.ALL`.
This should be moved to a configurable component in ESAPI.proprites and all of the LoggerFactories should be updated to use that value for generating new Logger instances.

### PR DESCRIPTION
**Pull request by sjoshid**
Fixes #631 
Creating this PR just for discussion. Feel free to close it if you think this is ridiculous.

### CODE REVIEW

File: src/main/java/org/owasp/esapi/logging/slf4j/Slf4JLogger.java [hunk 1]
```diff
@@ -49,11 +49,9 @@ private void log(int esapiLevel, EventType type, String message, Throwable throw
             logBridge.log(delegate, esapiLevel, type, message, throwable);
         }
     }
-    
 
     private boolean isEnabled(int esapiLevel) {
-        //Are Logger.OFF and Logger.ALL reversed?  This should be simply the less than or equal to check...
-        return (esapiLevel <= maxLogLevel && maxLogLevel != Logger.OFF) || maxLogLevel == Logger.ALL;
```
Comment (sjoshid) [inline src/main/java/org/owasp/esapi/logging/slf4j/Slf4JLogger.java:56]: This also means we dont need `maxLogLevel`

**Comment by jeremiahjstacey** [comment]
Although I agree with the intent of what's being requested, this is not the correct solution.  Although this does fulfill the need of issue #630, it will require rework should we address issue #631.

I do not believe this implementation is sustainable for that reason.  In addition, I'm fairly confident that since there are no test changes in this update that the baseline will currently not pass unit tests.
Please be sure you are running:

`mvn clean test` or `mvn clean package` before submitting PRs -- all tests must pass.

One other thing to note is that whatever change we make with SLF4J must be consistent with Log4J and Java Logger implementations. 

I feel strongly that my suggested solution (https://github.com/ESAPI/esapi-java-legacy/issues/630#issuecomment-881776104) is the most appropriate path given the full context of the discussion.

**Comment by jeremiahjstacey** [comment]
In my initial comment I missed we were trying to address #631. 
I was assuming we were fixing #630, my apologies.

We are not in a position to fix #631 at this time.  We need to correct the logic first, and negotiate the use cases presented by @kwwall  in #631 before we can know that we are making the intended changes.

I agree that the ESAPI library may need to expose configuration between the internal logging and the logging delegation, but at this time we have not had enough discussion to understand what the API contract should be at that boundary.

Although this is one possible approach, I'm not sure it covers all necessary cases -- I'm not really sure I fully understand all of the cases we're concerned with yet.

**Comment by sjoshid** [comment]
This was intended to fix both: #630 and #631.
> mvn clean test or mvn clean package before submitting PRs -- all tests must pass.

Yes. I just wanted to put forth my thoughts. Failing tests need to be changed for sure. One main thing with this PR is there is no change to downstream esapi clients. `maxLogLevel` was never exposed so it would work seamlessly. 

But I understand yours and Kevin concerns about this change. So I'll shut up now. 

Thanks for taking time to go through it.

**Comment by jeremiahjstacey** [comment]
Putting this on hold pending future discussions on #631.

**Pull request closed without merging**